data/mock/addressConverterMock.go

```diff
@@ -0,0 +1,66 @@
+package mock
+
+import (
+	"bytes"
+	"encoding/hex"
+	"errors"
+
+	"github.com/ElrondNetwork/elrond-go/data/state"
+)
+
+var errFailure = errors.New("failure")
+
+type AddressConverterMock struct {
+	Fail                                          bool
+	CreateAddressFromPublicKeyBytesRetErrForValue []byte
+}
+
+func (acm *AddressConverterMock) CreateAddressFromPublicKeyBytes(pubKey []byte) (state.AddressContainer, error) {
+	if acm.Fail {
+		return nil, errFailure
+	}
+
+	if acm.CreateAddressFromPublicKeyBytesRetErrForValue != nil {
+		if bytes.Equal(acm.CreateAddressFromPublicKeyBytesRetErrForValue, pubKey) {
```

Comment:
This is confusing. Use the typical function"Called" method from other mocks.

Reply:
this was mock copied over from process I think.